glpaper: init at unstable-2020-03-30

ccellado opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Add new useful package to the Wayland stack.

###### Things done

Added glpaper package.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+    sha256 = "04y12910wvhy4aqx2sa63dy9l6nbs7b77yqpdhc96x2b3mgzgjfs";
+  };
+
+  nativeBuildInputs = [ meson ninja pkgconfig ];
```
In new expressions, `pkg-config` should be used instead of the `pkgconfig` alias.

ccellado replied:
Done.

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+    libGL libglvnd
+  ];
+
+  enableParallelBuilding = true;
```
`enableParallelBuilding` is enabled by default in meson and cmake packages.

ccellado replied:
Done.

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+    description = "GLPaper is a wallpaper program for wlroots based wayland compositors such as sway that allows you to render glsl shaders as your wallpaper";
+    homepage    = "https://bitbucket.org/Scoopta/glpaper";
+    platforms   = platforms.linux;
+    maintainers = [ ];
```
License?

ccellado replied:
Done.

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+  meta = with stdenv.lib; {
+    description = "GLPaper is a wallpaper program for wlroots based wayland compositors such as sway that allows you to render glsl shaders as your wallpaper";
+    homepage    = "https://bitbucket.org/Scoopta/glpaper";
+    license = licenses.gpl3;
```
```suggestion
    license = licenses.gpl3Plus;
```

https://bitbucket.org/Scoopta/glpaper/src/a95db77088dfb5636a87f3743fc9b5eca70c1ae2/src/paper.c#lines-6:7

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  meta = with stdenv.lib; {
+    description = "GLPaper is a wallpaper program for wlroots based wayland compositors such as sway that allows you to render glsl shaders as your wallpaper";
+    homepage    = "https://bitbucket.org/Scoopta/glpaper";
```
Is the bitbucket a mirror of the sourcehut repo?

ccellado replied:
No info, but it looks like it.

ccellado replied:
Changed homepage to sourcehut.

ccellado commented:
Added myself to maintainers list.

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+    homepage    = "https://hg.sr.ht/~scoopta/glpaper";
+    license = licenses.gpl3Plus;
+    platforms   = platforms.linux;
+    maintainers = with maintainers; [ maintainers.ccellado ];
```
redundant 

```suggestion
    maintainers = with maintainers; [ ccellado ];
```

ccellado replied:
Done~

ccellado commented:
@jtojnar Anything else?

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+stdenv.mkDerivation {
+  name = "glpaper";
+  version = "a95db77088dfb5636a87f3743fc9b5eca70c1ae2";
```
This should be `unstable-$date` as per https://nixos.org/nixpkgs/manual/#sec-package-naming

ccellado replied:
Done.

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  nativeBuildInputs = [ meson ninja pkg-config ];
+  buildInputs = [
+    wlroots wayland wayland-protocols wlroots
```
There are few instances of weird spacing/line wrapping. Perhaps use nixpkgs-fmt.

jtojnar replied:
Also wlroots is here twice.

ccellado replied:
Fixed, thanks!

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+  ];
+
+  meta = with stdenv.lib; {
+    description = "GLPaper is a wallpaper program for wlroots based wayland compositors such as sway that allows you to render glsl shaders as your wallpaper";
```
```suggestion
    description = "Wallpaper program for wlroots based Wayland compositors such as sway that allows you to render glsl shaders as your wallpaper";
```

ccellado replied:
That's better, thank you.

jtojnar commented on pkgs/development/tools/glpaper/default.nix:
```diff
@@ -0,0 +1,35 @@
+  nativeBuildInputs = [ meson ninja pkg-config ];
+  buildInputs = [
+    wlroots wayland wayland-protocols wlroots
+    pixman libxkbcommon
```
Why does this need all these dependencies? I only see wayland-client, wayland-egl and egl in https://hg.sr.ht/~scoopta/glpaper/browse/meson.build?rev=default#L5

ccellado replied:
Yeah, I took that from overlay, did not test.

After testing removed :  pixman libxkbcommon libglvd.

taikatytto committed:
glpaper: init at unstable-2020-03-30

	modified:   maintainers/maintainer-list.nix
	new file:   pkgs/development/tools/glpaper/default.nix
	modified:   pkgs/top-level/all-packages.nix

ccellado reviewed (commented):
Messed up my commits when squashed them, how to remove there ones?

ccellado commented:
Got mixed up branches, recreated cleanly at #87953

Sorry for the inconvenience.

jtojnar commented:
In the future, you can use <code>git reset --hard *commit*</code> to change branch to point to a different commit.
